lib/internal/process/methods.js

```diff
@@ -8,11 +8,18 @@ const {
   validateMode,
   validateUint32
 } = require('internal/validators');
+const {
+  isMainThread
+} = require('internal/worker');
 
 function setupProcessMethods(_chdir, _cpuUsage, _hrtime, _memoryUsage,
                              _rawDebug, _umask, _initgroups, _setegid,
                              _seteuid, _setgid, _setuid, _setgroups) {
   // Non-POSIX platforms like Windows don't have certain methods.
+  // Workers also lack these methods since they change process-global state.
+  if (!isMainThread)
```

Comment:
nit: Perhaps put this check outside the call to `setupProcessMethods()` to avoid the small bit of call overhead ?

Reply:
I’d prefer not to because it seems conceivable that we would want other process methods that are not main-thread-only to be exposed from this function in the future, i.e. this way is a bit more readable to me